ocs-api/import-owner-private-keys2.sh

```diff
@@ -0,0 +1,300 @@
+    if [ "$(uname)" == "Darwin" ]; then
+      echo "Using macOS, will generate private key and certificate simultaneously."
+    else
+      openssl ecparam -genkey -name secp"${var2}"r1 -out "${keyType}"private-key.pem >/dev/null 2>&1
+      chk $? 'Generating an ecdsa private key.'
+    fi
+    keyCertGenerator
+  fi
+}
+
+function keyCertGenerator() {
+  if [[ -f "${keyType}"private-key.pem ]]; then
+    if [ "$(uname)" == "Darwin" ]; then
+      rm "${keyType}"private-key.pem
+    else
+      echo -e ""${keyType}" private key creation: Successful"
+      gencmd="openssl req -x509 -key "$privateKey" -days 3650 -out "$keyCert""
+    fi
+  fi
+  #Generate a private key and self-signed certificate.
+  #You should have these environment variables set. If they aren't you will be prompted to enter values.
+  #!/usr/bin/env bash
+  if [ "$(uname)" == "Darwin" ]; then
+    if [[ $keyType == "ecdsa256" ]] || [[ $keyType == "ecdsa384" ]]; then
+      (
+      echo "$COUNTRY_NAME"
+      echo "$STATE_NAME"
+      echo "$LOCALE_NAME"
+      echo "$COMPANY_NAME"
+      echo "$ORG_UNIT"
+      echo "$COMMON_NAME"
+      echo "$EMAIL_NAME"
+    ) | openssl req -x509 -nodes -days 3650 -newkey ec:<(openssl genpkey -genparam -algorithm ec -pkeyopt ec_paramgen_curve:P-"${var2}") -keyout "$privateKey" -out "$keyCert" >/dev/null 2>&1
+      chk $? 'generating ec certificate'
+      if [[ -f "$privateKey" ]]; then
+        openssl ec -in ecdsa"${var2}"private-key.pem -out ecdsa"${var2}"private-key.pem >/dev/null 2>&1
+        chk $? 'decrypting ec private key for macOS'
+      else
+        echo "No EC private key found"
+      fi
+    fi
+  fi
+  if [[ $keyType == "rsa" ]] || [ "$(uname)" != "Darwin" ]; then
+    (
+      echo "$COUNTRY_NAME"
+      echo "$STATE_NAME"
+      echo "$LOCALE_NAME"
+      echo "$COMPANY_NAME"
+      echo "$ORG_UNIT"
+      echo "$COMMON_NAME"
+      echo "$EMAIL_NAME"
+    ) | $gencmd >/dev/null 2>&1
+    chk $? 'generating rsa certificate'
+    if [ "$(uname)" == "Darwin" ] && [[ $keyType == "rsa" ]] && [[ -f "$privateKey" ]]; then
+        openssl rsa -in "$privateKey" -out "$privateKey" >/dev/null 2>&1
+        chk $? 'decrypting rsa private key for macOS'
+    fi
+  fi
+
+  if [[ -f $keyCert ]] && [[ -f "$privateKey" ]]; then
+    echo -e ""${keyType}" Private Key and "${keyType}"Key Certificate creation: Successful"
+    genPublicKey
+    popd >/dev/null
+  else
+    echo ""${keyType}" Private Key and "${keyType}"Key Certificate not found"
+    exit 2
+  fi
+}
+
+function genPublicKey() {
+  # This function is ran after the private key and owner certificate has been created. This function will create a public key to correspond with
+  # the owner private key/certificate. Generate a public key from the certificate file
+  openssl x509 -pubkey -noout -in $keyCert >"${keyType}"pub-key.pem
+  chk $? 'Creating public key...'
+  echo "Generating "${keyType}" public key..."
+  if [[ -f "${keyType}"pub-key.pem ]]; then
```

Comment:
Does the key ever not get generated when the openssl exit code is 0 ? Doesn't seem like this if stmt is necessary.